backend/planty/dashboard/views.py

```diff
@@ -376,3 +374,60 @@ def post(self, request: Request):
         event.save()
 
         return Response(status=status.HTTP_200_OK)
+
+
+class PlantView(APIView):
+    def get(self, request: Request, id):
+        user: User = request.user
+
+        try:
+            plant: Plant = Plant.objects.get(pk=id)
+        except Plant.DoesNotExist:
+            return Response(data={'id': ['Plant with the given ID does not exist']}, status=status.HTTP_404_NOT_FOUND)
+
+        if plant.user != user:
+            return Response(status=status.HTTP_403_FORBIDDEN)
+
+        plant_json = plant.values()
+        plant_json["events"] = []
+
+        # next watering event
+        planned_watering = max(plant.last_watered + timedelta(plant.instruction.watering), date.today())
+        days_late = max(0, (date.today() - planned_watering).days)
+        plant_json["events"].append({
+            'date': planned_watering,
+            'action': "water",
+            'days_late': days_late,
+            # 'priority': None,
+            'happened': False,
+            'custom_info': None
+        })
+        # next fertilizing event
+        planned_fertilizing = max(plant.last_fertilized + timedelta(plant.instruction.fertilizing), date.today())
+        days_late = max(0, (date.today() - planned_fertilizing).days)
+        plant_json["events"].append({
+            'date': max(plant.last_fertilized + timedelta(plant.instruction.fertilizing), date.today()),
+            'action': "fertilize",
+            'days_late': days_late,
+            # 'priority': None,
+            'happened': False,
+            'custom_info': None
+        })
+
+        try:
+            custom_events = CustomEvent.objects.filter(plant=plant)
+            custom_events = [{"date": event.date,
+                              'action': 'custom',
+                              # 'priority': None,
+                              'custom_info': {
+                                  'name': event.name,
+                                  'description': event.description}
+                              }
+                             for event in custom_events]
+
+            plant_json["custom_events"] = custom_events
+
+        except CustomEvent.DoesNotExist:
```

Comment:
Czy tutaj potrzebujemy tego? Jeśli nie znajdziemy nic, to to nie będzie raczej dla nas problem. Ja dawałem ten try except w przypadkach, gdy trzeba było zmodyfikować konkretny atrybut tego obiektu